discovery_engine_core/ai/ai/src/coi/key_phrase.rs

```diff
@@ -243,6 +243,7 @@ fn unify(
 /// The elements can be prepared before they are reduced. The reduced result can be finalized
 /// whereas the finalization conditially depends on whether the reduced lane overlaps with the main
 /// square block of the matrix.
+#[allow(clippy::needless_pass_by_value)] // arrayview instead of reference
```

Comment:
I don't understand the comment, we are getting an `ArrayBase` that could be an ArrayView but it could also be something else.

Reply:
this is because of a conflict between the clippy lint and the ndarray design. the function signature uses `ArrayBase` to accomodate for owned as well as viewed arrays. now clippy wants us to write it as `&ArrayBase`, but the whole concept of "referenced arrays" is tackled by views in ndarray. essentially, we could have a `&ArrayBase` which would eg be a `&Array` or `&ArrayView`, but the latter is just a redundant double reference in a sense.

i'm more in favor of following the ndarray design instead of the clippy lint in this case.

Reply:
I agree with you but I find the comment confusing since the signature does not request a view and an owned value can be passed in and in this case clippy is right.
I think a simple solution is to rephrase the comment, another is to actually request an ArrayView and modify the case in which we don't use it.

Reply:
i rephrased the comment